activerecord/test/cases/migration/references_foreign_key_test.rb

```diff
@@ -105,6 +106,18 @@ class ReferencesForeignKeyTest < ActiveRecord::TestCase
           @connection.remove_reference :testings, :testing_parent, foreign_key: true
         end
       end
+
+      test "foreign key column can be added for singular table" do
+        ActiveRecord::Base.stubs(:pluralize_table_names).returns(false)
+        @connection.create_table :testing
+        @connection.change_table :testing_parents do |t|
+          t.references :testing, foreign_key: true
+        end
+
+        fk = @connection.foreign_keys("testing_parents").first
+        assert_equal "testing_parents", fk.from_table
+        assert_equal "testing", fk.to_table
```

Comment:
Let's incorporate removing the key as well. I suspect this line needs to be changed in lockstep: https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L670